addons/project_purchase/models/project_project.py

```diff
@@ -12,32 +12,52 @@ class Project(models.Model):
     purchase_orders_count = fields.Integer('# Purchase Orders', compute='_compute_purchase_orders_count', groups='purchase.group_purchase_user', export_string_translation=False)
 
     def _compute_purchase_orders_count(self):
-        purchase_count_per_project = dict(
+        purchase_orders_per_project = dict(
             self.env['purchase.order']._read_group(
-                [('project_id', 'in', self.ids)],
-                ['project_id'], ['__count'],
+                domain=[('project_id', 'in', self.ids)],
+                groupby=['project_id'],
+                aggregates=['id:array_agg'],
             )
         )
-        for project in self:
-            project.purchase_orders_count = purchase_count_per_project.get(project)
+        purchase_orders_count_per_project_from_lines = dict(
+            self.env['purchase.order.line']._read_group(
+                domain=[
+                    ('order_id', 'not in', [order_id for values in purchase_orders_per_project.values() for order_id in values]),
```

Comment:
Maybe its because of Monday, but i had a hard time understanding this double for. Maybe changing 'values' into 'order_ids' could help with that ?

Reply:
For instance, `purchase_orders_per_project` may contain `{project.project(5,): [12, 13], project.project(1,): [1, 2]}` and I want to get all the values in one big list. So I think I have to double loop in order to do that :)